File: RecipeSharing-Angular/src/app/Models/Recipe.ts
import { User } from "./User";

export class Recipe {
  id!: number;
  name!: string;
  image!: string;
  userFk!: string;
  preparationTime!: number;
  mealType!: MealType;
  dietaryPreferences!: DietaryPreferences;
  user!: User;
  createdOn!: Date;
  recipeSteps: string[] = [];
  recipeAppliances: RecipeAppliance[] = [];
  recipeIngredients: RecipeIngredient[] = [];
}

export class RecipeStep {
  text!: string;
}

export class RecipeIngredient {
  measurementUnit!: MeasurementUnit;
  quantity!: number;
  ingredientName: string = '';
}

export class RecipeAppliance {
  ApplianceType!: ApplianceType;
}

export enum DietaryPreferences {
  Omnivore,
  Vegetarian,
  Vegan,
  Carnivore,
  Pescatarian,
  Keto,
  Paleo
};

export enum MeasurementUnit {
  Grams,
  Milligrams,
  Kilograms,
  Ounces,
  Liters,
  Milliliters,
  Teaspoons,
  Tablespoons,
  Cups
};

export enum MealType {
  Breakfast,
  Brunch,
  Lunch,
  Dinner,
  Dessert,
  Drink,
  Snack
};

export enum ApplianceType {
  Microwave,
  Blender,
  Processor,
  Toaster,
  RiceCooker,
  Juicer,
  Stove,
  Oven,
  AirFryer,
  DeepFryer,
  Mixer,
  Grill
};

File: RecipesSharing.Domain/Entities/Recipe.cs
namespace RecipesSharing.Domain.Entities
{
    using RecipesSharing.Domain.Enums;
    using System.ComponentModel.DataAnnotations.Schema;
    using static Microsoft.EntityFrameworkCore.DbLoggerCategory.Database;

    public class Recipe : BaseEntity
    {
        public string Name { get; set; }

        public string Image { get; set; }

        public string UserFk { get; set; }

        public int PreparationTime {  get; set; }

        public MealType MealType { get; set; }

        public DietaryPreferences DietaryPreferences { get; set; }

        [ForeignKey("UserFk")]
        public virtual AppUser? User { get; set; }

        public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

        public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();

        public virtual ICollection<RecipeAppliance> RecipeAppliances { get; set; } = new List<RecipeAppliance>();

        public virtual ICollection<RecipeIngredient> RecipeIngredients { get; set; } = new List<RecipeIngredient>();

        public virtual ICollection<RecipeStep> RecipeSteps { get; set; } = new List<RecipeStep>();
    }
}
